src/Swarm/Game/Step.hs

```diff
@@ -436,28 +438,22 @@ evalConst c vs k = do
       return $ Up (Fatal msg) k
     Right cek' -> return cek'
 
-
--- XXX load this from a file and have it available in a map?
---     Or make a quasiquoter (https://github.com/byorgey/swarm/issues/16) ?
--- XXX Remove the fromRight error case.
-
--- | A program to run a "seed" robot that regrows a harvested entity.
-seedProgram :: Text -> ProcessedTerm
-seedProgram thing = fromRight (error . from $ "Invalid program: " <> thing) prog
-  where
-    prog = processTerm . into @Text . unlines $
-      [ "let repeat : int -> cmd () -> cmd () = \\n.\\c."
-      , "  if (n == 0) {} {c ; repeat (n-1) c}"
-      , "in {"
-      , "  r <- random 500;"
-      , "  repeat (r + 100) wait;"
-      , "  appear \"|\";"
-      , "  r <- random 500;"
-      , "  repeat (r + 100) wait;"
-      , "  place \"" ++ from @Text thing ++ "\";"
-      , "  selfdestruct"
-      , "}"
-      ]
+-- | A system program for a "seed robot", to regrow a growable entity
+--   after it is harvested.
+seedProgram :: Text -> Either Text ProcessedTerm
+seedProgram thing = processParsedTerm [tmQ|
+  let repeat : int -> cmd () -> cmd () = \n.\c.
+    if (n == 0) {} {c ; repeat (n-1) c}
+  in {
+    r <- random 500;
+    repeat (r + 100) wait;
```

Comment:
not sure why, but inserting bad code here doesn't fail at compile time.

Reply:
It fails at compile time if there is a *parse* error.  That is all the quasiquoter does.  Typechecking and elaborating the term still happens at runtime (see the call to `processParsedTerm`). I thought a bit about how to make a quasiquoter that did the entire processing pipeline but it made my head hurt. But now that you mention it I want to try a little harder before merging this.

Reply:
Ask and ye shall receive!  With the latest commit it now it runs the entire processing pipeline at compile time, so it will fail at compile time on any kind of error, whether a parse error or type error.

Reply:
That's wonderful, thanks, my test is now properly failing.